python/ribasim/ribasim/model.py

```diff
@@ -449,4 +450,63 @@ def to_xugrid(self):
         uds = uds.assign_coords(from_node_id=(edge_dim, from_node_id))
         uds = uds.assign_coords(to_node_id=(edge_dim, to_node_id))
 
+        if add_results:
+            uds = self._add_results(uds)
+
+        return uds
+
+    def _add_results(self, uds):
+        toml_path = self.filepath
+        if toml_path is None:
+            raise FileNotFoundError("Model must be written to disk to add results.")
```

Comment:
Isn't `self.filepath` more of an implementation detail? Users would maybe try `model.filepath = "path/to/ribasim.toml"`, and then it would probably fail at the next check when the results don't exist.